obspy/io/dmx/core.py

```diff
@@ -0,0 +1,137 @@
+                                 ("localtime", np.int16),
+                                 ("datatype", "1S"),
+                                 ("descriptor", "1S"),
+                                 ("digi_by", np.int16),
+                                 ("processed", np.int16),
+                                 ("length", np.int32),
+                                 ("rate", np.float32),
+                                 ("mindata", np.float32),
+                                 ("maxdata", np.float32),
+                                 ("avenoise", np.float32),
+                                 ("numclip", np.int32),
+                                 ("timecorrect", np.double),
+                                 ("rate_correct", np.float32),
+                                  ])
+
+structtag_dtypes = np.dtype([("sinc", "1S"),
+                             ("machine", "1S"),
+                             ("id_struct", np.int16),
+                             ("len_struct", np.int32),
+                             ("len_data", np.int32)])
+
+types = {"s": ("uint16", "H", 2), "q": ("int16", "h", 2),
+         "u": ("uint16", "H", 2), "i": ("int16", "h", 2),
+         "2": ("int32", "i", 4), "l": ("int32", "i", 4),
+         "r": ("uint16", "H", 2), "f": ("float32", "f", 4),
+         "d": ("float64", "d", 8)}
+
+
+def readstructtag(fid):
+    y = AttribDict()
+    data = np.fromfile(fid, structtag_dtypes, 1)
+    for (key, (fmt, size)) in structtag_dtypes.fields.items():
+        if str(fmt).count("S") != 0:
+            y[key] = data[key][0].decode('UTF-8')
+        else:
+            y[key] = data[key][0]
+    return y
+
+
+def readdescripttrace(fid):
+    y = AttribDict()
+
+    data = np.fromfile(fid, descript_trace_dtypes, 1)
+
+    for (key, (fmt, size)) in descript_trace_dtypes.fields.items():
+        if str(fmt).count("S") != 0:
+            y[key] = data[key][0].decode('UTF-8')
+        else:
+            y[key] = data[key][0]
+
+    return y
+
+
+def readdata(fid, n, t):
+    target = types[t]
+    return np.fromfile(fid, eval("np.%s" % target[0]), n)
+
+
+def _is_dmx(filename):
+    try:
+        with open(filename, "rb") as fid:
+            while fid.read(12):  # we require at least 1 full structtag
+                fid.seek(-12, 1)
+                structtag = readstructtag(fid)
+                if structtag.id_struct == 7:
+                    descripttrace = readdescripttrace(fid)
+                    UTCDateTime(descripttrace.begintime)
+                    return True
+                else:
+                    fid.seek(
+                        int(structtag.len_struct) + int(structtag.len_data), 1)
+
+    except Exception:
+        return False
+    return True
+
+
+def _read_dmx(filename, **kwargs):
+    station = None
+    if "station" in kwargs:
+        station = kwargs["station"]
+
+    traces = []
+    with open(filename, "rb") as fid:
+        content = fid.read()
+
+    with SpooledTemporaryFile(mode='w+b') as fid:
```

Comment:
Huh, never seen that.. do I understand correctly that this will be in memory unless it can't fit in mem? Just asking because we really want this in mem if at all possible, but from docs that's what is happening?

Reply:
ah, indeed, that's actually because, at INGV, we are reading from the network, and reading in many little reads is awfully slow, so this is how I managed to read at once, then use the "fake file" that has proper ``read``, ``seek`` functions.

Reply:
Just asking, because we use `io.BytesIO` for stuff like this throughout the code base.. so if there's no particular reason to go a different route here, maybe we should keep it more consistent and use BytesIO? I think we can live with MemoryError in crazy use cases instead of disk swapping magic?

Reply:
It's just that I didn't manage to use ``io.BytesIO`` ... :(